packages/jaeger-ui/src/components/DeepDependencies/Header/index.css

```diff
@@ -28,8 +28,10 @@ limitations under the License.
 .DdgHeader--controlHeader {
   align-items: center;
   background-color: #fafafa;
+  border-top: 1px solid #eee;
   display: flex;
-  height: 40px; /* Arbitrary height until find is implemented */
+  flex-direction: row-reverse;
```

Comment:
without `row-reverse`, the UiFind element is right-aligned until a sibling renders, then it shoots over to the left side. With `row-reverse` it's consistently on the left side.

Reply:
The use of `row-reverse` is kind of counter-intuitive and, all things being equal, is better left avoided.

Why not move the wrapping `<div>` out of `HopsSelector` and into `Header`? This will prevent the issue you're referring to. The `HopsSelector` can return a `Fragment`?

Reply:
I updated `HopsSelector` to take `distanceToPathElems` as an optional param and to render an empty `div` when not provided. This way the distinction between rendering an empty `div` versus two selectors with popovers is encapsulated in `HopsSelector` and not a concern for `Header`